File: src/main/java/havarunner/exception/MemberIsNotPackagePrivateException.java
package havarunner.exception;

import java.lang.reflect.Member;
import java.lang.reflect.Modifier;

public class MemberIsNotPackagePrivateException extends CodingConventionException {
    public MemberIsNotPackagePrivateException(Member member) {
        super(String.format(
            "%s is %s. Please make it package private.",
            member.getName(),
            modifierToString(member)
        ));
    }

    private static String modifierToString(Member member) {
        return Modifier.isPrivate(member.getModifiers()) ? "private" :
            (Modifier.isPublic(member.getModifiers()) ? "public" : "protected");
    }
}

File: src/main/scala/havarunner/exception/MemberIsNotPackagePrivateException.scala
package havarunner.exception

import havarunner.exception.MemberIsNotPackagePrivateException._
import java.lang.reflect.{Modifier, Member}

class MemberIsNotPackagePrivateException(member: Member) extends CodingConventionException(
  modifierToString(member)
)

object MemberIsNotPackagePrivateException {
  def modifierToString(member: Member) =
    if (Modifier.isPrivate(member.getModifiers)) {
      "private"
    } else if (Modifier.isPublic(member.getModifiers)) {
      "public"
    } else if (Modifier.isProtected(member.getModifiers)) {
      "protected"
    } else {
      throw new RuntimeException("unrecognised modifier for " + member)
    }
}
